### ISSUE TITLE
BUILD: build issues on windows

### ISSUE THREAD

**Issue by jreback** [pydata/pandas#13721]
so from latest master w latest deps

- [x] closed by # 13736
```
======================================================================
ERROR: test_nanosecond_field (pandas.tseries.tests.test_timeseries.TestDatetime64)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\dev\pandas\pandas\tseries\tests\test_timeseries.py", line 3925, in test_nanosecond_field
    dti = DatetimeIndex(np.arange(10))
  File "C:\dev\pandas\pandas\util\decorators.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "C:\dev\pandas\pandas\tseries\index.py", line 337, in __new__
    if isinstance(subarr, DatetimeIndex):
UnboundLocalError: local variable 'subarr' referenced before assignment
```

- [ ] 
```
======================================================================
FAIL: test_next (pandas.io.tests.test_common.TestMMapWrapper)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\dev\pandas\pandas\io\tests\test_common.py", line 141, in test_next
    self.assertEqual(next_line, line)
AssertionError: 'a,b,c\r\n' != 'a,b,c\n'
- a,b,c
?      -
+ a,b,c

```

I think some @sinhrks changes for the first
not sure about the 2nd, IIRC @pijucha saw this on windows as well

@gfyoung any idea?d

**Comment by sinhrks** [comment]
1st issue is caused by a test case uses ``np.int32`` data (on windows). Will send a PR to fix the condition below.

- https://github.com/pydata/pandas/blob/master/pandas/tseries/index.py#L332

**Comment by gfyoung** [comment]
The second test is getting the content right, but it seems like an issue is what the newline is being read as at the end.  For all intensive purposes, can add ".strip()" before the comparison .

**Comment by pijucha** [comment]
As for the second test: 
On windows, I set up git to change newlines to `\r\n` in a working directory. So the file `test_mmap.csv` used in this test had `\r\n`. I noticed `readlines()` converted them to `\n` but `MMapwrapper` didn't.

**Comment by jreback** [comment]
closed by #13736 and #13732

**Comment by gfyoung** [comment]
@jreback : nit, check both boxes in the issue!
